src/utils/path.js

```diff
@@ -0,0 +1,34 @@
+const removeMultiSlashes = (path) => path.replace(/\/+/g, '/');
+const stripTrailingSlashes = (path) => path.replace(/\/+$/, '');
+const addLeadingSlash = (path) => path.charAt(0) === '/' ? path : '/' + path;
+const stripBasePath = (path, basePath) => basePath !== '/' && path.indexOf(basePath) === 0 ? path.substr(basePath.length) : path;
+
+export const parsePath = (path, options) => {
+  let pathname = stripBasePath(path, options.basePath);
+  let hash = '';
+  let anchor = null;
+
+  const hashIndex = pathname.indexOf('#');
+  if (hashIndex !== -1) {
```

Comment:
I've done this too, but now that I see it here, I think `hashIndex >= 0` would make more sense, otherwise it could also be `-2` or whatever. I know, totally irrelevant 😄